### ISSUE TITLE
False positives: local variables/parameters matching browser API names are flagged

### ISSUE THREAD

**Issue by Akshay090** [amilajack/eslint-plugin-compat#688]
## Problem

The `compat/compat` rule produces false positives when local variable or parameter names happen to collide with browser API names. Two related issues:

### 1. Arrow function / function expression parameters not recognized as local declarations

The `Identifier` visitor tracks locally declared names to suppress false positives when a name shadows a browser global. However, it only checks these parent node types: `Property`, `FunctionDeclaration`, `VariableDeclarator`, `ClassDeclaration`, `ImportDefaultSpecifier`, `ImportSpecifier`, `ImportDeclaration`.

Missing: `ArrowFunctionExpression` and `FunctionExpression`.

This means callback parameters that share a name with a browser API are flagged:

```js
// Flagged as "scheduler is not supported in Safari 26.2"
// but 'scheduler' here is a domain object, not Window.scheduler
schedulers.map(scheduler => scheduler.name);

// Same issue with function expressions
items.map(function(fetch) { return fetch.toString(); });
```

### 2. Early-return guard pattern not recognized as feature detection

The existing `isInsideIfStatement` check only suppresses errors for API usage **inside** an `if` block. The common early-return guard pattern is not detected:

```js
function setup() {
  if (!('serviceWorker' in navigator)) {
    return; // early exit if API unavailable
  }
  // This usage is safe but still flagged:
  navigator.serviceWorker.register('/sw.js');
}
```

## Impact

In a large codebase, these false positives can produce dozens of spurious errors, particularly when domain entities share names with newer browser APIs (e.g. `scheduler` matching the Prioritized Task Scheduling API `Window.scheduler`).

**Comment by Krinkle** [comment]
I've encounted the same regression recently. This passed until a few weeks ago, but is now failing. This is on a repo using `eslint-pulgin-compat: ^6.0.0`. From https://github.com/qunitjs/qunit/tree/3.0.0-rc1:

```js
var dump = {
    object: function (map, stack) {
        // ...
        for (let i = 0; i < keys.length; i++) {
          const key = keys[i];
          const val = map[key];
        }
      }
};
```
```
  6:23  error  Map is not supported in iOS Safari 7.0-7.1, IE 9  compat/compat
```
